cmd/cluster-node-tuning-operator/main.go

```diff
@@ -224,26 +224,24 @@ func removePerformanceOLMOperator(cfg *rest.Config) error {
 	}
 	csvSelector.Add(*req)
 	if err := k8sclient.List(context.TODO(), csvs, client.MatchingLabelsSelector{Selector: csvSelector}); err != nil {
-		if !errors.IsNotFound(err) {
+		if !errors.IsNotFound(err) && !util.IsNoMatchError(err) {
 			return err
 		}
 	}
 	for i := range csvs.Items {
 		csv := &csvs.Items[i]
 		deploymentSpecs := csv.Spec.InstallStrategy.StrategySpec.DeploymentSpecs
-		if deploymentSpecs != nil {
```

Comment:
was this "if" removed on purpose, if so why? if not perhaps it can be replaced with an "if dep.spec == nil" and then return with custom error.

Reply:
Oh yes, I removed it and left the reason in the squashed commit, it's a redundant check, the range in golang checks for nil and empty slices, so if it's ever nil or empty, the range will just not execute